src/test/kotlin/com/nftco/flow/sdk/TransactionTest.kt

```diff
@@ -107,23 +107,27 @@ class TransactionTest {
     fun `Can parse events`() {
         val accessApi = TestUtils.newMainnetAccessApi()
 
-        // https://flowscan.org/transaction/5e6ef76c524dd131bbab5f9965493b7830bb784561ca6391b320ec60fa5c395e
-        val tx = accessApi.getTransactionById(FlowId("5e6ef76c524dd131bbab5f9965493b7830bb784561ca6391b320ec60fa5c395e"))
+        // https://flowscan.org/transaction/8c2e9d37a063240f236aa181e1454eb62991b42302534d4d6dd3839c2df0ef14
+        val tx = accessApi.getTransactionById(FlowId("8c2e9d37a063240f236aa181e1454eb62991b42302534d4d6dd3839c2df0ef14"))
         assertThat(tx).isNotNull
 
-        val results = accessApi.getTransactionResultById(FlowId("5e6ef76c524dd131bbab5f9965493b7830bb784561ca6391b320ec60fa5c395e"))!!
-        assertThat(results.events).hasSize(4)
+        val results = accessApi.getTransactionResultById(FlowId("8c2e9d37a063240f236aa181e1454eb62991b42302534d4d6dd3839c2df0ef14"))!!
+        assertThat(results.events).hasSize(12)
         assertThat(results.events[0].event.id).isEqualTo("A.0b2a3299cc857e29.TopShot.Withdraw")
-        assertThat(results.events[1].event.id).isEqualTo("A.c1e4f4f4c4257510.Market.CutPercentageChanged")
-        assertThat(results.events[2].event.id).isEqualTo("A.0b2a3299cc857e29.TopShot.Deposit")
-
-        assertThat(results.events[3].event.id).isEqualTo("A.c1e4f4f4c4257510.Market.MomentListed")
-        assertThat("id" in results.events[3].event).isTrue
-        assertThat("price" in results.events[3].event).isTrue
-        assertThat("seller" in results.events[3].event).isTrue
-        assertThat("id" in results.events[3].event.value!!).isTrue
-        assertThat("price" in results.events[3].event.value!!).isTrue
-        assertThat("seller" in results.events[3].event.value!!).isTrue
+        assertThat(results.events[1].event.id).isEqualTo("A.0b2a3299cc857e29.TopShot.Deposit")
+        assertThat(results.events[2].event.id).isEqualTo("A.ead892083b3e2c6c.DapperUtilityCoin.TokensWithdrawn")
+
+        assertThat("from" in results.events[2].event).isTrue
+        assertThat("amount" in results.events[2].event).isTrue
+
+// TODO - fix this test complaining com.fasterxml.jackson.databind.exc.MismatchedInputException: Unknown CadenceType kind: Restriction
```

Comment:
Will follow up with a PR that re-enables this test as handling for Restriction type is included as part of these changes